sarwaveifrproc/utils.py

```diff
@@ -0,0 +1,198 @@
+        - model_intraburst (tf.keras.Model): Intraburst model loaded from the provided path.
+        - model_interburst (tf.keras.Model): Interburst model loaded from the provided path.
+        - scaler_intraburst (RobustScaler): Intraburst scaler loaded from the provided path.
+        - scaler_interburst (RobustScaler): Interburst scaler loaded from the provided path.
+        - bins_intraburst (dict): Dictionary containing intraburst bins for each predicted variable.
+        - bins_interburst (dict): Dictionary containing interburst bins for each predicted variable.
+    """    
+    # Unpack paths
+    path_model_intraburst, path_model_interburst, path_scaler_intraburst, path_scaler_interburst, path_bins_intraburst, path_bins_interburst = paths.values()
+    
+    # Load models and scalers using paths provided
+    model_intraburst = tf.keras.models.load_model(path_model_intraburst, compile=False)
+    scaler_intraburst_array = np.load(path_scaler_intraburst)
+    scaler_intraburst = RobustScaler(medians=scaler_intraburst_array[0], iqrs=scaler_intraburst_array[1])    
+    
+    model_interburst = tf.keras.models.load_model(path_model_interburst, compile=False)
+    scaler_interburst_array = np.load(path_scaler_interburst)
+    scaler_interburst = RobustScaler(medians=scaler_interburst_array[0], iqrs=scaler_interburst_array[1]) 
+    
+    bins_intraburst = {f'{var}': np.load(os.path.join(path_bins_intraburst, f'bins_{var}.npy')) for var in predicted_variables}
+    bins_interburst = {f'{var}': np.load(os.path.join(path_bins_interburst, f'bins_{var}.npy')) for var in predicted_variables}
+
+    return model_intraburst, model_interburst, scaler_intraburst, scaler_interburst, bins_intraburst, bins_interburst
+    
+    
+def process_files(input_safe, output_safe, model_intraburst, model_interburst, scaler_intraburst, scaler_interburst, bins_intraburst, bins_interburst, predicted_variables, product_id):
+    """
+    Processes files in the input directory, generates predictions, and saves results in the output directory.
+
+    Parameters:
+        input_safe (str): Input safe path.
+        output_safe (str): Path to the directory where output data will be saved.
+        model_intraburst (tf.keras.Model): Intraburst model for prediction.
+        model_interburst (tf.keras.Model): Interburst model for prediction.
+        scaler_intraburst (RobustScaler): Scaler for intraburst data.
+        scaler_interburst (RobustScaler): Scaler for interburst data.
+        bins_intraburst (dict): Dictionary containing intraburst bins for each predicted variable.
+        bins_interburst (dict): Dictionary containing interburst bins for each predicted variable.
+        predicted_variables (list): List of variable names to be predicted.
+        product_id (str): Identifier for the output product.
+
+    Returns:
+        None
+    """
+    subswath_filenames = glob.glob(os.path.join(input_safe, '*-vv-*.nc'))
```

Comment:
in the future L1B /L1B versions there will no longer have `'*-vv-*.nc'` sur probably `'-sv-` or `'-dv-'`. The processor would handle 2 cases.

Reply:
I modified  `'*-vv-*.nc'` to `'*-?v-*.nc'`